### PR TITLE
[Test] Ensure indices are healthy enough before closing

### LINKED ISSUE #99516: [CI] SnapshotStressTestsIT testRandomActivities failing
Looks different from https://github.com/elastic/elasticsearch/issues/99477

**Build scan:**
https://gradle-enterprise.elastic.co/s/oqy4rtnsvf3me/tests/:server:internalClusterTest/org.elasticsearch.snapshots.SnapshotStressTestsIT/testRandomActivities

**Reproduction line:**
```
./gradlew ':server:internalClusterTest' --tests "org.elasticsearch.snapshots.SnapshotStressTestsIT.testRandomActivities" -Dtests.seed=F11392CA75807FB3 -Dtests.locale=es-NI -Dtests.timezone=Pacific/Palau -Druntime.java=20
```

**Applicable branches:**
main

**Reproduces locally?:**
Didn't try

**Failure history:**
https://gradle-enterprise.elastic.co/scans/tests?tests.container=org.elasticsearch.snapshots.SnapshotStressTestsIT&tests.test=testRandomActivities

**Failure excerpt:**
```
[2023-09-13T17:20:06,538][INFO ][o.e.s.SnapshotsService   ] [node_s0] deleting snapshots [snapshot-partial-28] from repository [repo-0]
sep 13, 2023 5:20:06 PM com.carrotsearch.randomizedtesting.RandomizedRunner$QueueUncaughtExceptionsHandler uncaughtException
WARNING: Uncaught exception in thread: Thread[#8530,elasticsearch[node_s2][generic][T#5],5,TGRP-SnapshotStressTestsIT]
java.lang.AssertionError: max seq. no. [419] does not match [325]
	at __randomizedtesting.SeedInfo.seed([F11392CA75807FB3]:0)
	at org.elasticsearch.index.engine.ReadOnlyEngine.assertMaxSeqNoEqualsToGlobalCheckpoint(ReadOnlyEngine.java:194)
	at org.elasticsearch.index.engine.ReadOnlyEngine.ensureMaxSeqNoEqualsToGlobalCheckpoint(ReadOnlyEngine.java:180)
	at org.elasticsearch.index.engine.ReadOnlyEngine.<init>(ReadOnlyEngine.java:122)
	at org.elasticsearch.index.engine.NoOpEngine.<init>(NoOpEngine.java:63)
	at org.elasticsearch.index.engine.NoOpEngine.<init>(NoOpEngine.java:49)
	at org.elasticsearch.index.shard.IndexShard.innerOpenEngineAndTranslog(IndexShard.java:2005)
	at org.elasticsearch.index.shard.IndexShard.openEngineAndRecoverFromTranslog(IndexShard.java:1969)
	at org.elasticsearch.index.shard.StoreRecovery.lambda$internalRecoverFromStore$9(StoreRecovery.java:479)
	at org.elasticsearch.action.ActionListenerImplementations$ResponseWrappingActionListener.onResponse(ActionListenerImplementations.java:236)
	at org.elasticsearch.action.ActionListenerImplementations$DelegatingResponseActionListener.onResponse(ActionListenerImplementations.java:182)
	at org.elasticsearch.index.CompositeIndexEventListener.callListeners(CompositeIndexEventListener.java:278)
	at org.elasticsearch.index.CompositeIndexEventListener.iterateBeforeIndexShardRecovery(CompositeIndexEventListener.java:287)
	at org.elasticsearch.index.CompositeIndexEventListener.beforeIndexShardRecovery(CompositeIndexEventListener.java:314)
	at org.elasticsearch.index.shard.IndexShard.preRecovery(IndexShard.java:1681)
	at org.elasticsearch.index.shard.StoreRecovery.internalRecoverFromStore(StoreRecovery.java:410)
	at org.elasticsearch.index.shard.StoreRecovery.recoverFromStore(StoreRecovery.java:92)
	at org.elasticsearch.index.shard.IndexShard.recoverFromStore(IndexShard.java:2338)
	at org.elasticsearch.action.ActionRunnable$3.doRun(ActionRunnable.java:73)
	at org.elasticsearch.common.util.concurrent.ThreadContext$ContextPreservingAbstractRunnable.doRun(ThreadContext.java:983)
	at org.elasticsearch.common.util.concurrent.AbstractRunnable.run(AbstractRunnable.java:26)
	at java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1144)
	at java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:642)
	at java.base/java.lang.Thread.run(Thread.java:1623)
```

### PR DESCRIPTION
**Pull request by ywangd**
Closing unassigned indices is not cleanly handled. This change avoids doing so in the test.

Resolves: #99516

### CODE REVIEW

File: server/src/internalClusterTest/java/org/elasticsearch/snapshots/SnapshotStressTestsIT.java [hunk 1]
```diff
@@ -842,8 +856,8 @@ private void startSnapshotter() {
                 boolean startedSnapshot = false;
                 try (TransferableReleasables localReleasables = new TransferableReleasables()) {
 
-                    // separate TransferableReleasables for blocking node restarts & index deletion so we can release these blocks and
-                    // permit data node restarts and index deletions as soon as the snapshot starts
+                    // separate TransferableReleasables for blocking node restarts so we can release these blocks and
+                    // permit data node restarts as soon as the snapshot starts
```
Comment (ywangd) [inline server/src/internalClusterTest/java/org/elasticsearch/snapshots/SnapshotStressTestsIT.java:860]: IIUC, we do block index deletion for the whole process of taking full snapshots. It is not blocked when taking partial ones. Hence the change.

**Review by DaveCTurner — CHANGES_REQUESTED** [review]
Index deletion is ok I think, it's _closing_ an unhealthy index (before restoring over it) that is the problem.

**Comment by ywangd** [comment]
You are absolutely right. I did study your [sample failure](https://github.com/elastic/elasticsearch/issues/99516#issuecomment-1761298380) on the issue and I was aware this is for "closing" an index. Somehow it came out as fixing for "deleting". Sorry for the mistake.

**Review by DaveCTurner — APPROVED** [review]
I see, sorry, I stopped after reading the title :)

I was thinking this wouldn't be quite enough, we also need to block node restarts until the close-indices step is complete. But now that I look more closely I see we already block node restarts in (the lambda created in) `startRestorer()`. So yes LGTM.

**Pull request merged**